clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/JustificationProcessor.java

```diff
@@ -30,30 +32,43 @@
 import com.google.protobuf.Value;
 import com.google.protobuf.util.JsonFormat;
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.codec.binary.Base64;
 
+@Slf4j
 @AllArgsConstructor(onConstructor = @__({@Inject}))
-public class JustificationProcessor {
+public class JustificationProcessor implements LogMutator {
   private static final String JUSTIFICATION_LOG_METADATA_KEY = "justification";
   private final JvsClient jvs;
 
   /**
-   * Validates the given {@code jvsToken} and populates the {@link AuditLogRequest} with the
+   * Extracts and validates the {@code jvsToken} and populates the {@link AuditLogRequest} with the
    * justification info.
    *
-   * @param jvsToken A JWT produced by JVS.
    * @param auditLogRequest Audit log request to be processed.
    * @return The {@link AuditLogRequest} with justification filled.
    * @throws LogProcessingException when it fails to populate the justification info.
    */
-  public AuditLogRequest process(String jvsToken, AuditLogRequest auditLogRequest)
-      throws LogProcessingException {
+  @Override
+  public AuditLogRequest process(AuditLogRequest auditLogRequest) throws LogProcessingException {
     Preconditions.checkArgument(
         auditLogRequest.hasPayload(), "audit log request doesn't have payload");
 
     AuditLogRequest.Builder auditLogRequestBuilder = auditLogRequest.toBuilder();
     AuditLog.Builder auditLogBuilder = auditLogRequest.getPayload().toBuilder();
-    auditLogBuilder = this.auditLogBuilderWithJustification(jvsToken, auditLogBuilder);
+
+    Value jvsToken =
+        auditLogRequest
+            .getContext()
+            .getFieldsMap()
+            .get(AuditLoggingServerInterceptor.JUSTIFICATION_TOKEN_HEADER_KEY);
+    if (jvsToken == null || "".equals(jvsToken.getStringValue())) {
```

Comment:
Similar to above comment, how about `jvsToken.getStringValue().isEmpty()` instead?